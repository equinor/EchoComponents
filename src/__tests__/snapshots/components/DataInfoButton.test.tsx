import React from 'react';
import renderer from 'react-test-renderer';
import { DataInfoButton } from '../../..';

it('renders open DataInfoButton with badge correctly', () => {
    const data = {
        numberOfItems: 5,
        label: 'P&ID',
        ariaLabel: 'P&ID',
        onTagInfoClicked: jest.fn(),
        itemType: 'document'
    };

    const tree = renderer.create(<DataInfoButton data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders open DataInfoButton without badge correctly', () => {
    const data = {
        numberOfItems: 1,
        label: 'D&ID',
        onTagInfoClicked: jest.fn(),
        itemType: 'document'
    };

    const tree = renderer.create(<DataInfoButton data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders open DataInfoButton with badge with over 99 items correctly', () => {
    const data = {
        numberOfItems: 120,
        label: 'Others',
        onTagInfoClicked: jest.fn(),
        itemType: 'document'
    };

    const tree = renderer.create(<DataInfoButton data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
});
