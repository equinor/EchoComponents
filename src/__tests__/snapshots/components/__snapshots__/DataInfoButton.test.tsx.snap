// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders open DataInfoButton with badge correctly 1`] = `
<button
  aria-label="P&ID"
  className="sc-dlfnbm bmmLHw undefined button buttonWithBadge"
  data-count={5}
  disabled={false}
  onClick={[MockFunction]}
  tabIndex={0}
  type="button"
>
  <span
    className="sc-gsTCUz dRDfTG"
  >
    <span
      className="sc-bdfBwQ bdhdma"
    >
      P&ID
    </span>
  </span>
</button>
`;

exports[`renders open DataInfoButton with badge with over 99 items correctly 1`] = `
<button
  aria-label="Others"
  className="sc-dlfnbm bmmLHw undefined button buttonWithBadge"
  data-count="99+"
  disabled={false}
  onClick={[MockFunction]}
  tabIndex={0}
  type="button"
>
  <span
    className="sc-gsTCUz dRDfTG"
  >
    <span
      className="sc-bdfBwQ bdhdma"
    >
      Others
    </span>
  </span>
</button>
`;

exports[`renders open DataInfoButton without badge correctly 1`] = `
<button
  aria-label="D&ID"
  className="sc-dlfnbm bmmLHw undefined button "
  data-count={1}
  disabled={false}
  onClick={[MockFunction]}
  tabIndex={0}
  type="button"
>
  <span
    className="sc-gsTCUz dRDfTG"
  >
    <span
      className="sc-bdfBwQ bdhdma"
    >
      D&ID
    </span>
  </span>
</button>
`;
